tap_pardot/client.py

```diff
@@ -85,7 +88,13 @@ def _make_request(self, method, url, params=None):
             url,
             params,
         )
-        response = requests.request(
+
+        request_session = requests.Session()
+        retries = Retry(total=100, backoff_factor=1, status_forcelist=[502, 503, 504])
+        request_session.mount("http://", HTTPAdapter(max_retries=retries))
+        request_session.mount("https://", HTTPAdapter(max_retries=retries))
```

Comment:
actually, aren't we already doing retries with the `backoff` decorator? This seems redundant..